### PR TITLE
Fix potential RCE

### PR DESCRIPTION
**Pull request by LaserEyess**
See #134 for the discussion. I'd like to fix this "the right way", but this will do for now.

### CODE REVIEW

**Comment by LaserEyess** [comment]
A function like this should really have tests, I think.

File: src/remote-exec.c [hunk 1]
```diff
@@ -186,11 +186,15 @@ gchar *build_remote_exec_cmd(TrgClient * tc, GtkTreeModel * model,
             }
 
             if (valuestr) {
-                tmp = g_regex_replace(replacerx, work, -1, 0, valuestr, 0,
+                // Values are not guaranteed to be shell safe
+                shell_tmp = g_shell_quote(valuestr);
```
Comment (TingPing) [inline src/remote-exec.c:190]: More clear variable name like `valueEscaped`?

**Comment by LaserEyess** [comment]
@TingPing as a quick fix I'm 80% sure this is right, the one thing I'm not 100% sure of is this part of `g_shell_quote()`'s docs:

>The quoting style used is undefined (single or double quotes may be used).

I am guessing "undefined" here should mean that "`g_shell_quote()` will pick the more appropriate quote type depending on the string contents", and not "it may get it wrong sometimes or behave poorly on a string with mixed quotes".

**Comment by TingPing** [comment]
Hmm, I'm not sure. I can maybe look into this weekend.

**Comment by LaserEyess** [comment]
From `#gtk`
> - does this mean "glib will choose the more appropriate quoting style depending on the string" or does it mean "glib may get it wrong if there are mixed quotations"
> - no, it means "it'll alwas be correct, but when a better method is found, glib might change what it does"

**Pull request merged**